libsolidity/analysis/NameAndTypeResolver.cpp

```diff
@@ -62,8 +62,7 @@ bool NameAndTypeResolver::registerDeclarations(SourceUnit& _sourceUnit, ASTNode
 	}
 	catch (langutil::FatalError const&)
 	{
-		if (m_errorReporter.errors().empty())
-			throw; // Something is weird here, rather throw again.
+		solAssert(!m_errorReporter.errors().empty());
```

Comment:
What's the reason for changing this? Technically, the previous approach rethrows the exception and thus provides more information than just asserting.

Reply:
Another factor was that this is quite unlikely to happen (we rarely just throw `FatalError`, we use the helpers that automatically report an error) so I also wanted to make this kind of check simpler. Someone who added it originally clearly though it wasn't self-explanatory enough and felt the need to add a comment. An assert is IMO pretty self-explanatory.

Reply:
Yeah, it doesn't matter too much in the end, fine to just keep this change, if you prefer it that way.

Reply:
ok. I still added the message to the assert. This will make things clearer if it does happen.